File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> bracket;
        for(char c : s){
            if(c=='(' || c=='[' || c=='{'){
                bracket.push(c);
            }else{
                if(bracket.empty() || !isMatching(bracket.top(),c)){
                    return false;
                }
                bracket.pop();
            }
        }
        return bracket.empty();
    }
    private:
    bool isMatching(char open, char close){
        return (open=='(' && close==')' || open=='[' && close==']' || open=='{' && close=='}');
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        for(char i : s.toCharArray()){
            if(i=='(' || i=='{' || i=='['){
                stack.push(i);
                continue;
            }
            else if(stack.isEmpty() || !isMatching(i, stack.peek())){
                return false;
            }
            stack.pop();
        }
        return stack.isEmpty();
    }
    public boolean isMatching(char i, char top){
        return (i=='}' && top=='{') || (i==']' && top=='[') || (i==')' && top=='(');
    }
}